server/handlers_test.go

```diff
@@ -67,6 +68,49 @@ func TestHTTPServer_ReportEndpoint(t *testing.T) {
 	}, testTimeout)
 }
 
+// TODO: test more cases for rule endpoint
+func TestHTTPServer_RuleEndpoint(t *testing.T) {
+	helpers.RunTestWithTimeout(t, func(t testing.TB) {
+		defer helpers.CleanAfterGock(t)
+
+		helpers.GockExpectAPIRequest(t, helpers.DefaultServicesConfig.AggregatorBaseEndpoint, &helpers.APIRequest{
+			Method:   http.MethodGet,
+			Endpoint: ira_server.RuleEndpoint,
+			EndpointArgs: []interface{}{
+				testdata.OrgID,
+				testdata.ClusterName,
+				testdata.UserID,
+				fmt.Sprintf("%v|%v", testdata.RuleErrorKey1.RuleModule, testdata.RuleErrorKey1.ErrorKey),
+			},
+		}, &helpers.APIResponse{
+			StatusCode: http.StatusOK,
+			Body:       testdata.Report3SingleRuleExpectedResponse,
+		})
+
+		helpers.GockExpectAPIRequest(t, helpers.DefaultServicesConfig.ContentBaseEndpoint, &helpers.APIRequest{
+			Method:   http.MethodGet,
+			Endpoint: ics_server.AllContentEndpoint,
+		}, &helpers.APIResponse{
+			StatusCode: http.StatusOK,
+			Body:       helpers.MustGobSerialize(t, testdata.RuleContentDirectory3Rules),
+		})
+
+		go content.RunUpdateContentLoop(helpers.DefaultServicesConfig)
```

Comment:
how is this supposed to work?

Reply:
Same as here https://github.com/RedHatInsights/insights-results-smart-proxy/pull/211/files#diff-e5b0e65ddaf27a82498a9a19be9839c2L54 . It's updating rules content (groups, templates and etc) by timer.